src/style.js

```diff
@@ -172,6 +174,29 @@ export function applyGroupedChannelStyles(selection, {target}, {ariaLabel: AL, t
   applyTitleGroup(selection, T);
 }
 
+// clip: true clips to the frame
+// TODO: accept other types of clips (paths, urls, x, y, other marks?…)
+// https://github.com/observablehq/plot/issues/181
+export function maybeClip(clip) {
+  if (clip === true) return "frame";
+  if (clip == null || clip === false) return false;
+  throw new Error(`clip method not implemented: ${clip}`);
+}
+
+export function applyClip(selection, mark, {width, height, marginLeft, marginRight, marginTop, marginBottom}) {
+  if (mark.clip === "frame") {
+    const id = `plot-clippath-${++nextClipPathId}`;
```

Comment:
We’ll need to reindex these in the test framework so that the tests are independent. You should be able to mirror what we did for markers which have similar behavior.